Kameleoon updated their HackMD doc - updated docs accordingly (#2254)
segmentio/segment-docs: src/connections/destinations/catalog/kameleoon/index.md

rewrite: true
---

> [Kameleoon](https://kameleoon.com/en) GDPR, CCPA, and HIPAA compliant, Kameleoon’s powerful and easy-to-use A/B testing, full stack, and AI-powered personalization solutions help marketers, product owners, and developers maximize customer engagement and conversion - all from a single platform.

Reviewer: @charholt why was the word 'provides' taken out? This intro sentence reads really strange

Reviewer: ```suggestion
[Kameleoon’s](https://kameleoon.com/en) powerful and easy-to-use A/B testing, full stack, and AI-powered personalization solutions help marketers, product owners, and developers maximize customer engagement and conversion all from a single platform.
```

Author: It is strange, and it was taken out.

* [Kameleoon AI Personalization](https://www.kameleoon.com/en/platform/personalization){:target="_blank"} (Web Client-side)

> First, you will have to request an API Key in order to use Kameleoon & Segment.io integration. You can request it to [support@kameleoon.com](mailto:support@kameleoon.com), by specifying your Kameleoon account login.

Reviewer: ```suggestion
Before adding Kameleoon as a destination, you first need to request an API Key in order to use the Kameleoon & Segment.io integration. Send your request it to [support@kameleoon.com](mailto:support@kameleoon.com), and be sure to specify your Kameleoon account login.
```

You will also need the **sitecode** key. You can find it in your Kameleoon account, as described in [How do I find my site id?](https://help.kameleoon.com/question/how-do-i-find-my-site-id/){:target="_blank"}.

To add the destination to your Segment workspace:

1. From the Destinations catalog page in the Segment App, click **Add Destination**.

> 2. Search for "Kameleoon" in the Destinations Catalog, and select the "Kameleoon" destination.

Reviewer: ```suggestion
2. Search for **Kameleoon** in the Destinations Catalog, and select the **Kameleoon** destination.
```

> 3. Choose which Source should send data to the "Kameleoon" destination.

Reviewer: ```suggestion
3. Choose which Source should send data to the **Kameleoon** destination.
```

4. Enter the **API Key** and the **sitecode** in the Kameleoon destination settings in Segment.

## Implementation pre-requisites: matching Users between Kameleoon and Segment

The integration requires you to use the same system of identifiers for both tools, meaning the `userId` value you pass to Segment should be the same value as Kameleoon uses to identify a “visitor”.

> If you use Kameleoon Experiment or Kameleoon AI Personalization on your website, Kameleoon places a cookie that contains an anonymous unique identifier, called the Kameleoon visitorcode, randomly assigned to a visitor. This ID is used to uniquely identify a browser.

Reviewer: ```suggestion
If you use Kameleoon Experiment or Kameleoon AI Personalization on your website, Kameleoon places a cookie that contains an anonymous unique identifier, called the Kameleoon visitorcode. This ID is randomly assigned to a visitor and is used to uniquely identify a browser.
```

> You can either pass our visitorcode in the userId property of the Segment calls to ensure Kameleoon can consolidate data and avoid any analytics discrepancies between the tools in your campaign results, OR you may specify in all Segment calls an additional user property, `k_visitorCode`, whose value is the Kameleoon visitorcode. You can retrieve the user visitorCode from the browser by using our [Activation API](https://developers.kameleoon.com/activation-api.html#visitor){:target="_blank"}. `Kameleoon.API.Visitor.code`

Reviewer: ```suggestion
You can either pass Kameleoon's visitorcode in the userId property of the Segment calls to ensure Kameleoon can consolidate data and avoid any analytics discrepancies between the tools in your campaign results, OR you may specify in all Segment calls an additional user property, `k_visitorCode`, whose value is the Kameleoon visitorcode. You can retrieve the user visitorCode from the browser by using Kameleoon's [Activation API](https://developers.kameleoon.com/activation-api.html#visitor){:target="_blank"}, `Kameleoon.API.Visitor.code`
```

Reviewer: @charholt why was `Kameleoon.API.Visitor.code` added to the end of the paragraph without any context?

Author: Yes - let me connect with Kameleoon and then address your questions

> If you are not able to update your existing Segment tracking plan, you can also pass our visitor code in an additional Segment call as follow:`analytics.track('Kameleoon identifier', {'k_visitorCode': Kameleoon.API.Visitor.code});Kameleoon` will automatically link your internal user ID or anonymous ID to our own visitorCode. Make sure to call it only once per session.

Reviewer: ```suggestion
If you are not able to update your existing Segment tracking plan, you can also pass Kameleoon's visitor code in an additional Segment call as follow:`analytics.track('Kameleoon identifier', {'k_visitorCode': Kameleoon.API.Visitor.code});Kameleoon`. This will automatically link your internal user ID or anonymous ID to Kameleoon's own visitorCode. Make sure to call it only once per session.
```

> If you use Kameleoon Full Stack, please refer to the [Kameleoon SDK documentation](https://developers.kameleoon.com/sdks-overview.html){:target="_blank"}. We always recommend to use your own internal ID to identify users.

Reviewer: ```suggestion
If you use Kameleoon Full Stack, please refer to the [Kameleoon SDK documentation](https://developers.kameleoon.com/sdks-overview.html){:target="_blank"}. Kameleoon recommends you to use your own internal ID to identify users.
```
